riak_test/yz_solr_upgrade_downgrade.erl

```diff
@@ -0,0 +1,459 @@
+        end
+    ),
+    %%
+    %% None of the indices we have tweaked should be available,
+    %% except for index-a1, which we haven't touched.
+    %%
+    yz_rt:wait_for_solr(Cluster),
+    yz_rt:wait_for_index(Cluster, ?INDEX_A1),
+    ?assertEqual(
+        lists:duplicate(5, false),
+        [rpc:call(Node1, yz_index, exists, [Index]) ||
+            Index <- [?INDEX_A2, ?INDEX_B1, ?INDEX_B2, ?INDEX_B3, ?INDEX_C]]
+    ),
+    %%
+    %% Finish the downgrade by reverting the luceneMatchVersion and
+    %% deleting the index data for all indices (including index-a!)
+    %%
+    rt:stop(Node1),
+    [{params, Params}] = ets:lookup(DowngradeData, params),
+    finish_downgrade(Params),
+    rt:start(Node1),
+    %%
+    %% Wait for Solr and indices to come up
+    %%
+    yz_rt:wait_for_solr(Cluster),
+    [yz_rt:wait_for_index(Cluster, Index) ||
+        Index <- [?INDEX_A1, ?INDEX_A2, ?INDEX_B1, ?INDEX_B2, ?INDEX_B3, ?INDEX_C]
+    ],
+    %%
+    %% add Node1 back to the set of candidates for cover plans,
+    %% and turn YZ AAE back on.
+    %%
+    lager:info("Re-enabling dist_query on Node1 and turning on AAE ... "),
+    rpc:call(Node1, yz_solr_proc, set_dist_query, [true]),
+    rpc:call(Node1, application, set_env, [?YZ_APP_NAME, anti_entropy, {on, []}]),
+    %%
+    %% Wait for a full round of AAE
+    %%
+    yz_rt:wait_for_full_exchange_round(Cluster, erlang:now()),
+    %%
+    %% Verify that all missing Solr data has been repaired
+    %% (Note that we never added more data to index-a1)
+    %%
+    verify_data(Cluster, ?INDEX_A1, 100),
+    [verify_data(Cluster, Index, 200) ||
+        Index <- [?INDEX_A2, ?INDEX_B1, ?INDEX_B2, ?INDEX_B3, ?INDEX_C]
+    ],
+    %%
+    %% Write and verify the next 100 entries to all of our buckets,
+    %% including the newly created index-c.
+    %%
+    write_and_verify_data(Cluster, ?BUCKET_A1, ?INDEX_A1, 101, 100, 200),
+    rt:pmap(
+        fun({Bucket, Index}) ->
+            ok = write_and_verify_data(Cluster, Bucket, Index, 201, 100, 300)
+        end,
+        [{?BUCKET_A2, ?INDEX_A2},
+         {?BUCKET_B1, ?INDEX_B1},
+         {?BUCKET_B2, ?INDEX_B2},
+         {?BUCKET_B3, ?INDEX_B3},
+         {?BUCKET_C,  ?INDEX_C}]
+    ),
+    pass.
+
+write_data(Cluster, Bucket, Start, End) ->
+    ValueGenerator = fun(_Key) ->
+        {"Same as it ever was", "text/plain"}
+    end,
+    lists:all(
+        fun(Result) -> Result =:= ok end,
+        yz_rt:write_data(Cluster, Bucket, Start, End, ValueGenerator)
+    ).
+
+
+verify_data(Cluster, Index, NumKeys) ->
+    ok = yz_rt:verify_num_match(yokozuna, Cluster, Index, NumKeys),
+    ExpectedSolrMatch = (?N_VAL * NumKeys * length(Cluster)) div ?CLUSTER_SIZE,
+    ok = yz_rt:verify_num_match(solr, Cluster, Index, ExpectedSolrMatch),
+    ok.
+
+augment_config(ApplicationKey, Key, Value, Config) ->
+    ApplicationConfig = proplists:get_value(ApplicationKey, Config),
+    ApplicationDict = dict:from_list(ApplicationConfig),
```

Comment:
Why convert things to and from dict here instead of just working with the proplist directly?